docker_ci_deploy/__main__.py

```diff
@@ -9,25 +9,82 @@
 from itertools import chain
 
 
-def strip_image_tag(image):
+def split_image_tag(image_tag):
     """
-    Remove the tag part from a Docker tag and return the rest. Full tags are of
-    the form [REGISTRYHOST/][NAME/...]NAME[:TAG] where the REGISTRYHOST may
-    contain a ':' but no '/', the NAME parts may contain '/' but no ':', and
-    the TAG part may contain neither ':' nor '/'.
+    Split the given image tag into its name and tag parts (<name>[:<tag>]).
+
+    Full tags are of the form [REGISTRYHOST/][NAME/...]NAME[:TAG] where the
+    REGISTRYHOST may contain a ':' but no '/', the NAME parts may contain '/'
+    but no ':', and the TAG part may contain neither ':' nor '/'.
     """
-    match = re.match('^((?:[^\/]+\/)?[^:]+)(?::[^:\/]+)?$', image)
+    match = re.match(r'^((?:[^\/]+\/)?[^:]+)(?::([^:\/]+))?$', image_tag)
     if match is None:
-        raise RuntimeError('Unable to parse tag "%s"' % (image,))
+        raise ValueError("Unable to parse image tag '%s'" % (image_tag,))
+
+    return match.group(1), match.group(2)
+
+
+def join_image_tag(image, tag):
+    """ Join an image name and tag. """
+    if tag is None or not tag:
+        return image
+
+    return ':'.join((image, tag))
+
+
+def replace_image_registry(image, registry):
+    if registry is None:
+        return image
 
-    return match.group(1)
+    return _join_image_registry(_strip_image_registry(image), registry)
 
 
-def strip_image_registry(image):
+def _strip_image_registry(image):
     # TODO: Not yet sure how to differentiate REGISTRYHOST from a NAME segment.
```

Comment:
No. Here's the grammar: https://github.com/docker/distribution/blob/master/reference/reference.go

There wasn't a nice summary before (there was just code) so I might have another try at implementing the proper regexes. Before, I could not find a way to definitively separate the registry address from the other name components.